crates/test-utils/src/lib.rs

```diff
@@ -0,0 +1,210 @@
+//! Common utilities for the integration tests, these are intended to make it
+//! easier to write complex tests of the compiler's functionality.
+
+use std::{fmt::Write, path::Path};
+
+use anyhow::bail;
+use cairo_lang_runner::{
+    casm_run::format_next_item,
+    Arg,
+    RunResultValue,
+    SierraCasmRunner,
+    StarknetState,
+};
+use cairo_lang_sierra::program::Program;
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+use hieratika_errors::compile::llvm::Error;
+use inkwell::{
+    context::Context,
+    memory_buffer::MemoryBuffer,
+    values::GenericValue,
+    OptimizationLevel,
+};
+
+/// Executes the LLVM-IR program and returns the exit code as unsigned integer.
+///
+/// # Arguments
+///
+/// - `path` - The path of the LLVM-IR file to execute
+/// - `function_name` - The name of the function to execute
+/// - `args` - The arguments to call the function.
+///
+/// # Errors
+///
+/// - [`anyhow::Error`] if the path does not exist.
+///
+/// # Panics
+///
+/// - It panics if the function name does not exist.
+pub fn execute_llvm_ir(
+    path: &Path,
+    function_name: &str,
+    args: &[&GenericValue],
+) -> anyhow::Result<u64> {
+    let context = Context::create();
+    let memory_buffer = MemoryBuffer::create_from_file(path).map_err(Error::from)?;
+    let module = context.create_module_from_ir(memory_buffer).map_err(Error::from)?;
+    let execution_engine = module
+        .create_jit_execution_engine(OptimizationLevel::None)
+        .map_err(Error::from)?;
```

Comment:
Does this create an LLI session, or a no-optimization ORC JIT? We specifically want LLI here, as it does not involve compilation to machine code for execution.

Reply:
Ahhh ok! I thought it was best to use the JIT engine for interpreted execution.